File: lib/layer2_core/light/e_on_secondary_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnSecondaryLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl26;
      case 'standard': return cstl26;
      case 'blue': return cbl26;
      case 'green': return cgl26;
      case 'red': return crl26;
      case 'nothing': return cnol26;
      default: return cstl26;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnSecondaryLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl26
            AppTheme.STANDARD -> R.color.cstl26
            AppTheme.BLUE -> R.color.cbl26
            AppTheme.GREEN -> R.color.cgl26
            AppTheme.RED -> R.color.crl26
            AppTheme.NOTHING -> R.color.cnol26
        }
    }
}
